apis/vmware/v1beta1/vspheremachine_types.go

```diff
@@ -64,6 +65,22 @@ type VSphereMachineSpec struct {
 	// Volumes is the set of PVCs to be created and attached to the VSphereMachine
 	// +optional
 	Volumes []VSphereMachineVolume `json:"volumes,omitempty"`
+
+	// PowerOffMode describes the desired behavior when powering off a VM.
+	//
+	// There are three, supported power off modes: hard, soft, and
+	// trySoft. The first mode, hard, is the equivalent of a physical
+	// system's power cord being ripped from the wall. The soft mode
+	// requires the VM's guest to have VM Tools installed and attempts to
+	// gracefully shut down the VM. Its variant, trySoft, first attempts
+	// a graceful shutdown, and if that fails or the VM is not in a powered off
+	// state after reaching 5 minutes timeout, the VM is halted.
+	//
+	// If omitted, the mode defaults to trySoft.
+	//
+	// +optional
+	// +kubebuilder:default=trySoft
+	PowerOffMode vmopv1.VirtualMachinePowerOpMode `json:"powerOffMode,omitempty"`
```

Comment:
Q: Is the reason we don't have the timeout for this CRD because the vmoperator has the timeout baked in? (and so we can't set it on the vmoperator CRD)

Reply:
Maybe we want to open an issue in the vm-operator repo (if it doesn't exist already) so they consider exposing this (and if yes a corresponding one in CAPV to adapt once they support it)

Reply:
I believe that's an issue filed to this https://github.com/vmware-tanzu/vm-operator/issues/179